### ISSUE TITLE
[Azure-Identity] How to atexit DefaultAzureCredential?

### ISSUE THREAD

**Issue by NetGH** [Azure/azure-sdk-for-python#8972]
How to gracefully dispose DefaultAzureCredential?
```python
import asyncio
from azure.identity.aio import DefaultAzureCredential

async def test():
    credential = DefaultAzureCredential()
    await credential.get_token('https://management.core.windows.net/.default')
    
asyncio.run( test() )
```

Warning:
=============================================
Unclosed client session
client_session: <aiohttp.client.ClientSession object at 0x00000253F1779688>
Unclosed connector
connections: ['[(<aiohttp.client_proto.ResponseHandler object at 0x00000253F1788438>, 114171.5)]']
connector: <aiohttp.connector.TCPConnector object at 0x00000253F1779748>
Unclosed client session
client_session: <aiohttp.client.ClientSession object at 0x00000253F1754888>

**Comment by chlowell** [comment]
Thanks for reporting this! I just opened an issue about it as well (#8990) before seeing this 😆. Currently there's no graceful way to close the session through the credential's public API. I'll fix it internally so credentials properly close their own sessions.

**Comment by chlowell** [comment]
I found no acceptable way for credentials to close their own transport sessions, so in the next release async credentials will be async context managers with `close` methods (implemented by #9090). That creates two ways to close a credential's session in your example:

1. Call `close`
```py
import asyncio
from azure.identity.aio import DefaultAzureCredential

async def test():
    credential = DefaultAzureCredential()
    await credential.get_token('https://management.core.windows.net/.default')
    await credential.close()
    
asyncio.run( test() )
```

2. Use `async with`
```py
import asyncio
from azure.identity.aio import DefaultAzureCredential

async def test():
    credential = DefaultAzureCredential()
    async with credential:
        await credential.get_token('https://management.core.windows.net/.default')
    
asyncio.run( test() )
```

**Comment by chlowell** [comment]
The methods I described above are available now in [azure-identity 1.2.0](https://pypi.org/project/azure-identity/1.2.0/). I'll close this issue for now; thank you again for opening it. Please reopen this, or open another issue, if you have further trouble.
